code/datums/helper_datums/teleport.dm

```diff
@@ -118,15 +118,21 @@
 		teleatom.forceMove(destturf)
 		playSpecials(destturf,effectout,soundout)
 		var/atom/impediment
+		var/valid = 0
 
 		if(destturf.density)
 			impediment = destturf
 
 		else
 			for(var/atom/movable/A in destturf)
 				if(A.density && A.anchored)
-					impediment = A
-					break
+					if(A.flags & ON_BORDER)

```

Comment:
This can be rewritten to something like if(!(A.flags & ON_BORDER) || prob(10))

Reply:
That would make the logic a bit unintuitive, but it would save a few lines. I don't think it's required to change it, though.

Reply:
It's not logically equivalent, Burger.

Reply:
I believe it is.